ampd-handlers/src/bin/solana/handler.rs

```diff
@@ -0,0 +1,981 @@
+use std::collections::HashMap;
+use std::fmt;
+use std::fmt::Debug;
+
+use ampd::handlers::solana_verify_msg::Message;
+use ampd::handlers::solana_verify_verifier_set::VerifierSetConfirmation;
+use ampd::monitoring;
+use ampd::solana::msg_verifier::verify_message;
+use ampd::solana::verifier_set_verifier::verify_verifier_set;
+use ampd::solana::{SolanaRpcClientProxy, SolanaTransaction};
+use ampd_handlers::voting::{self, Error, PollEventData as _, VotingHandler};
+use ampd_sdk::event::event_handler::{EventHandler, SubscriptionParams};
+use ampd_sdk::grpc::client::EventHandlerClient;
+use async_trait::async_trait;
+use axelar_wasm_std::chain::ChainName;
+use axelar_wasm_std::msg_id::Base58SolanaTxSignatureAndEventIndex;
+use axelar_wasm_std::voting::{PollId, Vote};
+use cosmrs::AccountId;
+use error_stack::{Report, ResultExt};
+use events::{try_from, AbciEventTypeFilter, Event, EventType};
+use serde::Deserialize;
+use solana_sdk::pubkey::Pubkey;
+use solana_sdk::signature::Signature;
+use tracing::info;
+use typed_builder::TypedBuilder;
+
+pub type Result<T> = error_stack::Result<T, Error>;
+pub type DomainSeparator = [u8; 32];
+
+#[derive(Clone, Debug, Deserialize)]
+#[try_from("wasm-messages_poll_started")]
+pub struct MessagesPollStarted {
+    poll_id: PollId,
+    source_chain: ChainName,
+    #[serde(deserialize_with = "ampd::solana::deserialize_pubkey")]
+    source_gateway_address: Pubkey,
+    confirmation_height: u64,
```

Comment:
Solana doesn't need `confirmation_height`, right? you can remove it here and pass `None` to the voting:PollStartedEvent